pallets/xcm/src/lib.rs

```diff
@@ -112,8 +115,11 @@ impl<T: Config> DmpMessageHandler for UnlimitedDmpExecution<T> {
 		let mut used = 0;
 		for (_sent_at, data) in iter {
 			let id = sp_io::hashing::twox_64(&data[..]);
-			let msg = VersionedXcm::<T::Call>::decode(&mut &data[..])
-				.map(Xcm::<T::Call>::try_from);
+			let msg = VersionedXcm::<T::Call>::decode_and_advance_with_depth_limit(
```

Comment:
Here we don't really want to advance the data? We more want to decode *all* the data, otherwise it should be an error?

Reply:
Ah ok, so `decode`'s default behaviour is to advance the cursor, but if we wanna change the behaviour to "decode everything otherwise error`, we can change that too.